File: leetcode/problems/1.go
func twoSum(nums []int, target int) []int {
  m := map[int]int{}
  for idx, num := range nums {
    if v, ok := m[target - num]; ok {
      return []int{v, idx}
    }
    m[num] = idx
  }
  return nil
}

File: leetcode/problems/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function (nums, target) {
  const map = new Map()
  let ans = []
  for (let [index, val] of nums.entries()) {
    if (map.has(target - val)) {
      ans = [index, map.get(target - val)]
      break
    }
    map.set(val, index)
  }
  return ans
};
